implementations/rust/ockam/ockam_command/src/tcp/connection/delete.rs

```diff
@@ -26,43 +21,20 @@ pub struct DeleteCommand {
 
 impl DeleteCommand {
     pub fn run(self, options: CommandGlobalOpts) {
-        let cfg = &options.config;
-        let node =
-            extract_address_value(&self.node_opts.api_node).unwrap_or_else(|_| "".to_string());
-        let port = cfg.get_node_port(&node).unwrap();
-        connect_to(port, self, delete_connection);
+        node_rpc(run_impl, (options, self))
     }
 }
 
-pub async fn delete_connection(
-    ctx: Context,
-    cmd: DeleteCommand,
-    mut base_route: Route,
-) -> anyhow::Result<()> {
-    let resp: Vec<u8> = match ctx
-        .send_and_receive(
-            base_route.modify().append(NODEMANAGER_ADDR),
-            api::delete_tcp_connection(&cmd)?,
-        )
-        .await
-    {
-        Ok(sr_msg) => sr_msg,
-        Err(e) => {
-            eprintln!("Wasn't able to send or receive `Message`: {}", e);
-            std::process::exit(exitcode::IOERR);
-        }
-    };
-    let r: Response = api::parse_response(&resp)?;
+async fn run_impl(
+    ctx: ockam::Context,
+    (options, command): (CommandGlobalOpts, DeleteCommand),
+) -> crate::Result<()> {
+    let node_name = extract_address_value(&command.node_opts.api_node)?;
+    let mut rpc = Rpc::background(&ctx, &options, &node_name)?;
+    let request = Request::delete("/node/tcp/connection")
+        .body(DeleteTransport::new(&command.id, command.force));
+    rpc.request(request).await?;
 
-    match r.status() {
-        Some(Status::Ok) => println!("Tcp connection `{}` successfully delete", cmd.id),
-        _ => {
-            eprintln!("Failed to delete tcp connection");
-            if !cmd.force {
-                eprintln!("You may have to provide --force to delete the API transport");
-                std::process::exit(exitcode::UNAVAILABLE);
-            }
-        }
-    }
+    println!("Tcp connection `{}` successfully delete", command.id);
```

Comment:
We had a typo here, could you fix it while we are here please? `delete` -> `deleted`